Dockerfile

```diff
@@ -113,674 +115,686 @@ RUN set -x; \
+	&& cd $MW_HOME/extensions/HeaderFooter \
+	&& git checkout -q 579df9effa112c45e5d83cd8e4ee052a37c15343 \
+	# SimpleMathJax add Fix path to ext.SimpleMathJax.js in ResourceModules \
+	&& rm -fr $MW_HOME/extensions/SimpleMathJax \
+	&& git clone --single-branch -b master https://github.com/WikiTeq/SimpleMathJax.git $MW_HOME/extensions/SimpleMathJax \
+	&& cd $MW_HOME/extensions/SimpleMathJax \
+	&& git checkout -q 1ef413553dca4143294842fac99b56425d815396 \
+#	# SocialProfile breaks Special:SpecialPages, see WIK-702?focusedCommentId=41955
+#	&& cd $MW_HOME/extensions/SocialProfile \
+#	&& git fetch origin master \
+#	&& git checkout -q 3ba3b1808d2ba5d48470a6258eb6e716ccc5473a \
+	# Throws the errors, see WIK-702?focusedCommentId=41196
+	&& rm -fr $MW_HOME/extensions/NumerAlpha \
+	# does not work? see WIK-702?focusedCommentId=41955
+	&& rm -fr $MW_HOME/extensions/TimedMediaHandler \
+	# see WLDR-242
+	&& cd $MW_HOME/extensions/PageForms \
+	&& git checkout -q fb9511cd59845b9d2e5bbeb2964a5c4fca698c13 \
+	# missed in Canasta
+	&& cd $MW_HOME/extensions/EmailAuthorization \
+	&& git submodule update --init --recursive
 
-# GoogleAnalyticsMetrics: Resolve composer conflicts, so placed before the composer install statement!
-COPY _sources/patches/core-fix-composer-for-GoogleAnalyticsMetrics.diff /tmp/core-fix-composer-for-GoogleAnalyticsMetrics.diff
+# Patch composer
 RUN set -x; \
-	cd $MW_HOME \
-	&& git apply /tmp/core-fix-composer-for-GoogleAnalyticsMetrics.diff
+	sed -i 's="monolog/monolog": "2.2.0",="monolog/monolog": "^2.2",=g' $MW_HOME/composer.json
 
 # GoogleLogin: removed explicit monolog/monolog v2 dependency
 COPY _sources/patches/GoogleLogin.monolog.diff /tmp/GoogleLogin.monolog.diff
 RUN set -x; \
-    cd $MW_HOME/extensions/GoogleLogin \
-    && git apply /tmp/GoogleLogin.monolog.diff
+	cd $MW_HOME/extensions/GoogleLogin \
+	&& git apply /tmp/GoogleLogin.monolog.diff
 
+# WikiTeq AL-12
 COPY _sources/patches/FlexDiagrams.0.4.fix.diff /tmp/FlexDiagrams.0.4.fix.diff
 RUN set -x; \
-    cd $MW_HOME/extensions/FlexDiagrams \
-    && git apply /tmp/FlexDiagrams.0.4.fix.diff
-
-# Fix composer dependencies for MassPasswordReset extension
-# TODO: remove when PR merged https://github.com/nischayn22/MassPasswordReset/pull/1
-COPY _sources/patches/MassPasswordReset.patch /tmp/MassPasswordReset.patch
-RUN set -x; \
-	cd $MW_HOME/extensions/MassPasswordReset \
-	&& git apply /tmp/MassPasswordReset.patch
+	cd $MW_HOME/extensions/FlexDiagrams \
+	&& git apply /tmp/FlexDiagrams.0.4.fix.diff
 
 # Composer dependencies
-COPY _sources/configs/composer.canasta.json $MW_HOME/composer.local.json
+# Original Canasta string:
+# COPY _sources/configs/composer.canasta.json $MW_HOME/composer.local.json
+# Modified by WikiTeq
+# COPY _sources/configs/composer.canasta.json $MW_HOME/composer.canasta.json
+COPY _sources/configs/composer.wikiteq.json $MW_HOME/composer.local.json
```

Comment:
I am not very sure about this composer file rename, it's good to have it as `composer.wikiteq.json`, yes, but at the same time if we'd like to pull some updates from Canasta upstream in future this rename will cause issues. So despite it's good looking thing I'd vote to stay practical and keep the original name

Reply:
their composer.canasta.json breaks things and one time was useless at all. Initially the idea was to add our stuff as composer.wikiteq.json to their composer.canasta.json, but the guys love to install broken extensions or break something especially using composer.canasta.json, so I get rid of the one...

Reply:
I see, this makes sense then, I don't think we should try to inherit everything from Canasta, I actually think opposite